GeneralJS/balanced-parens-2.ts

```diff
@@ -16,6 +16,91 @@
  *
  */
 
-function balancedParens2(input): boolean {}
+// solve parens 1 first lol
+
+function balancedParens2(input) {
+    /*
+        ((())[]{}) = true
+        var hubble = function() { telescopes.awesome();) = false
+    */
+   
+    // breaking case : closed symbol is proceeded by a none matching open symbol
+
+    // define regex
+    const parsingRegex = /[\[\]\{\}\(\)]/g;
+    const openRegex = /[\[\{\(]/g
+    const closedRegex = /[\]\}\)]/g
+    // parens only
+    const parsedInput = input.match(parsingRegex)
```

Comment:
Consider the time complexity of parsing out the parens and the balanced-parens simplified answer (how many times do you really need to iterate through the input string?